velox/functions/lib/string/StringImpl.h

```diff
@@ -197,38 +197,110 @@ std::vector<int32_t> stringToCodePoints(const T& inputString) {
   return codePoints;
 }
 
-/// Returns the starting position in characters of the Nth instance(counting
-/// from the left if lpos==true and from the end otherwise) of the substring in
-/// string. Positions start with 1. If not found, 0 is returned. If subString is
-/// empty result is 1.
+/// Returns the starting position in characters of the Nth instance of the
+/// substring in string after 'startPosition'. Positions start with 1. If not
+/// found, 0 is returned. If subString is empty result is 1.
+/// @tparam lpos If true, counting from the left and from the end otherwise.
+/// @param instance The instance of the substring to find in string.
+/// @param startPosition The starting character position in the string
+/// to search for the substring.
 template <bool isAscii, bool lpos = true, typename T>
-FOLLY_ALWAYS_INLINE int64_t
-stringPosition(const T& string, const T& subString, int64_t instance = 0) {
+FOLLY_ALWAYS_INLINE int64_t stringPosition(
+    const T& string,
+    const T& subString,
+    int64_t instance = 0,
+    std::optional<int64_t> startPosition = std::nullopt) {
```

Comment:
Wondering if it makes sense to replace (string, startPosition) arguments with a single std::string_view argument.

Reply:
Thanks for the suggestion. Updated to use std::string_view argument type.